Include/internal/pycore_ceval_state.h

```diff
@@ -14,49 +14,38 @@ extern "C" {
 
 typedef int (*_Py_pending_call_func)(void *);
 
-struct _pending_call {
+typedef struct _pending_call {
     _Py_pending_call_func func;
     void *arg;
     int flags;
-};
-
-#define PENDINGCALLSARRAYSIZE 300
+    struct _pending_call *next;
+} _pending_call;
 
-#define MAXPENDINGCALLS PENDINGCALLSARRAYSIZE
 /* For interpreter-level pending calls, we want to avoid spending too
    much time on pending calls in any one thread, so we apply a limit. */
-#if MAXPENDINGCALLS > 100
-#  define MAXPENDINGCALLSLOOP 100
-#else
-#  define MAXPENDINGCALLSLOOP MAXPENDINGCALLS
-#endif
+#define MAXPENDINGCALLSLOOP 100
 
-/* We keep the number small to preserve as much compatibility
-   as possible with earlier versions. */
-#define MAXPENDINGCALLS_MAIN 32
```

Comment:
This point is still valid, so we should probably leave nearly all of the "main" pending calls stuff alone.